sdk/core/azure-core/inc/azure/core/logging/logging.hpp

```diff
@@ -9,143 +9,51 @@
 
 #pragma once
 
-#include "azure/core/dll_import_export.hpp"
-
 #include <functional>
-#include <initializer_list>
-#include <set>
 #include <string>
-#include <utility>
 
 namespace Azure { namespace Core { namespace Logging {
-  class LogClassification;
-  class LogClassifications;
+  /**
+   * @brief Log message level.
+   */
+  enum class LogLevel
+  {
+    /// Logging level for failures that the application is unlikely to recover from.
+    Error,
+
+    /// Logging level when a function fails to perform its intended task.
+    Warning,
+
+    /// Logging level when a function operates normally.
+    Informational,
+
+    /// Logging level for detailed troubleshooting scenarios.
+    Verbose,
+  };
 
   /**
    * @brief Defines the signature of the callback function that application developers must write in
    * order to receive Azure SDK log messages.
    *
-   * @param classification The log message classification.
-   * @param classification The log message.
+   * @param level The log message level.
+   * @param message The log message.
    */
-  typedef std::function<void(LogClassification const& classification, std::string const& message)>
-      LogListener;
+  typedef std::function<void(LogLevel level, std::string const& message)> LogListener;
 
   /**
    * @brief Set the function that will be invoked to report an SDK log message.
    *
    * @param logListener A #LogListener function that will be invoked when the SDK reports a log
-   * message matching one of the log classifications passed to #SetLogClassifications(). If null, no
+   * message matching one of the log levels passed to #SetLogLevel(). If `nullptr`, no
    * function will be invoked.
    */
   void SetLogListener(LogListener logListener);
 
   /**
-   * @brief Allows the application to specify which log classification types it is interested in
-   * receiving.
+   * @brief Sets the maximul #LogLevel an application is interested in receiving.
+   * @detail Default is #LogLevel::Verbose.
```

Comment:
```suggestion
   * @detail Default is #LogLevel::Informational.
```

Default should be informational

Reply:
Or default should be none.

Reply:
it is "None" in a sense that you get none until you set callback. If you set it, you are probably interested in at least errors?
I set Verbose, because I would expect to get, simply, all, when I set logger. Then if I get too much, I tune it down, but not vice versa - I may know about the `SetLogListener()` but not about `SetLogLevel()`. Then I don't get enough log messages, so I think the SDK is not instrumented well enough with log messages.

Reply:
Also, other languages don't have `LogLevel::None`

Reply:
To give us some room, I am simply leaving the default undocumented now, without changing the actual default value. I still think Verbose is the best default choice, in terms of user-friendliness.